### PR TITLE
fix(ingest): various avro codegen fixes

### PR DESCRIPTION
**Pull request by hsheth2**
Lots of stuff in this PR:
- Fix generated types for enum
- Fix serialization and deserialization of unambiguous union types
- Add a validate method to each generated class
- Add a top-level command to validate the schemas of JSON MCE files
- Make all non-nullable types required in the generated class init methods
- Use the new schema_pattern option to streamline the MySQL integration test
- Add a script to update the MCE golden files for integration testing

See also: https://github.com/hsheth2/avro_gen/compare/9e6c1269e2df065a3dfa2d1bda4db7443ae7031f...master.
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-ingestion/.gitignore [hunk 1]
```diff
@@ -4,6 +4,7 @@ output
 src/datahub/metadata/
 pvenv36/
 bq_credentials.json
+/tmp
```
Comment (shirshanka) [inline metadata-ingestion/.gitignore:7]: did you mean tmp/ ?

File: metadata-ingestion/src/datahub/entrypoints.py [hunk 2]
```diff
@@ -77,10 +78,27 @@ def ingest(config: str):
...
+
+@datahub.command(context_settings=DEFAULT_CONTEXT_SETTINGS)
+@click.option(
+    "-f",
+    "--json-file",
+    type=click.Path(exists=True, dir_okay=False),
+    help="JSON file with a list of MCE objects",
+    required=True,
+)
+def check_mce_schema(json_file: str):
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/entrypoints.py:100]: It seems very specific to add a check_mce_schema command to the top-level `datahub` cli. 

Could we achieve the same outcome (validate that inputs look good) by adding a --dry-run mode to ingest ? 

So `datahub ingest -c mce_to_console.yml --dry-run`

File: metadata-ingestion/.gitignore (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/.gitignore:7]: The leading slash refers to the current directory in gitignore syntax, so this should be fine

File: metadata-ingestion/src/datahub/entrypoints.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/entrypoints.py:100]: Yep agreed that it probably doesn't deserve a top-level command

What should `--dry-run` do exactly - perhaps it should just set the sink to `console`?

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**